openwisp_controller/config/tests/test_admin.py

```diff
@@ -975,6 +975,28 @@ def test_preview_device_405(self):
         response = self.client.get(path, {})
         self.assertEqual(response.status_code, 405)
 
+    def test_config_error_reason(self):
+        device = self._create_device(name='download')
+        config = self._create_config(device=device)
+        url = reverse(f'admin:{self.app_label}_device_change', args=[device.pk])
+
+        with self.subTest('Test config status "modified" or "applied"'):
+            self.assertEqual(config.status, 'modified')
+            response = self.client.get(url)
+            self.assertNotContains(response, '<label>Error reason:</label>')
+            config.set_status_applied()
+            response = self.client.get(url)
+            self.assertNotContains(response, '<label>Error reason:</label>')
+
+        with self.subTest('Test config status "error"'):
+            config.set_status_error(reason='Reason not reported by the device.')
+            response = self.client.get(url)
+            self.assertContains(response, '<label>Error reason:</label>')
+            self.assertContains(
+                response,
+                '<div class="readonly">Reason not reported by the device.</div>',
+            )
```

Comment:
Let's get used to pass `html=True` to `assertContains`, which helps if the HTML being checked is a bit more complex than that.